tests/integration/test_charm.py

```diff
@@ -43,13 +43,48 @@ async def test_build_and_deploy(ops_test: OpsTest):
     assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
 
 
+async def test_seldon_istio_relation(ops_test: OpsTest):
+    """Test Seldon/Istio relation."""
+    # NOTE: This test is re-using deployment created in test_build_and_deploy()
+
+    istio_gateway = "istio-ingressgateway"
+    istio_pilot = "istio-pilot"
+    await ops_test.model.deploy(
+        entity_url="istio-gateway",
+        application_name=istio_gateway,
+        channel="latest/edge",
+        config={"kind": "ingress"},
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        istio_pilot,
+        channel="latest/edge",
+        config={"default-gateway": "test-gateway"},
+        trust=True,
+    )
+    await ops_test.model.add_relation(istio_pilot, istio_gateway)
+
+    await ops_test.model.wait_for_idle(
+        apps=[istio_pilot], status="active", raise_on_blocked=True, timeout=60 * 10
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[istio_gateway], status="active", raise_on_blocked=True, timeout=60 * 10
+    )
```

Comment:
Can we merge these to a single wait?
```suggestion
    await ops_test.model.wait_for_idle(
        apps=[istio_pilot, istio_gateway], status="active", raise_on_blocked=True, timeout=60 * 10
    )
```

Reply:
I wanted to make sure `istio-pilot` is up before `istio-gateway`. Will this do the same?

Reply:
Ok if you're making sure one is up before the other then no, the way I propose wont achieve that.  

It shouldn't matter (pilot and gateway can be deployed and start up in either order), but if it does matter then no they cannot be merged

Reply:
Pilot seem to startup for quite a long time. Sometimes gateway was timeout before pilot was up.

Reply:
I'm not sure what you mean by gateway being timed out.  Gateway should deploy correctly and go into Waiting until it gets the data from the istio-pilot relation.  You might be describing a bug!

Reply:
When gateway is in waiting state it is not in active state. So, while it waits for Istio pilot to come up the wait might expire for gateway.
I guess I might have observed what I see in integration tests in CI: Istio Pilot never getting IP. So, the reasoning was to wait till Isio pilot is up and then check gateway. Looks like there could be some other issue.

Reply:
Gateway's waiting for pilot has no timeout mechanism, if we leave it there forever it'll happily wait.  So I'm pretty sure it is safe to just merge these into a single wait

Reply:
Sounds good.